include/rocksdb/manifest_ops.h

```diff
@@ -0,0 +1,30 @@
+//  Copyright (c) 2011-present, Facebook, Inc.  All rights reserved.
+//  This source code is licensed under both the GPLv2 (found in the
+//  COPYING file in the root directory) and Apache 2.0 License
+//  (found in the LICENSE.Apache file in the root directory).
+
+#pragma once
+#include <cassert>
+
+#include "rocksdb/env.h"
+#include "rocksdb/file_checksum.h"
+#include "rocksdb/status.h"
+
+namespace ROCKSDB_NAMESPACE {
+
+Status GetCurrentManifestPath(const std::string& abs_path, FileSystem* fs,
+                              bool is_retry, std::string* manifest_path,
+                              uint64_t* manifest_file_number);
+
+// Requires absolute path to CURRENT file directory as a parameter.
+// Returns the FileChecksumList constructed purely off of the MANIFEST file
+// that was considered 'current' at the time of invoking this API.
+//
+// NOTE: We're NOT locking the referenced MANIFEST-* file.
+// It might become stale in case of intertwining DB operations.
+Status GetFileChecksumsFromCurrentManifest(Env* src_env,
+                                           const std::string& abs_path,
+                                           uint64_t manifest_file_size,
```

Comment:
Not clear how to provide/handle this parameter

Reply:
Per our offline conversation, given that caller likely doesn't have the context about the underlying MANIFEST file (or doesn't want to assume it), I'm going to get rid of this parameter and simply infer 'the current' MANIFEST file size from the file system at the time of calling the API.